### PR TITLE
f/resource_aws_mq_broker: Adds validation to broker_name and security_groups arguments in resource_aws_mq_broker

### LINKED ISSUE #17966: MQ Broker Name and Security Group validations
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

According to [AWS MQ Docs](https://docs.aws.amazon.com/amazon-mq/latest/developer-guide/amazon-mq-limits.html) the `name` and `security_groups` field are missing validations

### New or Affected Resource(s)

* aws_mq_broker

### PR DESCRIPTION
**Pull request by kubawi**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #17966

It's my first contribution to terraform-provider-aws. I tried my best to follow the contribution guides, but please let me know if I missed something. As far as the code change goes, I followed the patterns I could notice in the codebase, but any feedback is welcome.

Output from acceptance testing:

```
➜ make testacc TESTARGS='-run=TestValidateMQBrokerName'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestValidateMQBrokerName -timeout 180m
=== RUN   TestValidateMQBrokerName
--- PASS: TestValidateMQBrokerName (0.00s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1.551s
```

### CODE REVIEW

File: aws/validators.go [hunk 1]
```diff
@@ -2427,6 +2427,11 @@ var validateCloudWatchEventArchiveName = validation.All(
 	validation.StringMatch(regexp.MustCompile(`^[\.\-_A-Za-z0-9]+`), ""),
 )
 
+var validateMQBrokerName = validation.All(
+	validation.StringLenBetween(1, 50),
+	validation.StringMatch(regexp.MustCompile(`^[0-9A-Za-z._~-]+$`), ""),
```
Comment (gdavison) [inline aws/validators.go:2432]: When I've tried in the AWS Console, it doesn't accept `~` or `.`

**Review by gdavison — CHANGES_REQUESTED** [review]
Thanks for the PR, @kubawi, it looks great! I have one comment about the regex used for validation. It would also be good to update the documentation to reflect the new validation.

You can make the suggested changes, or I can make them and merge your contribution.

File: aws/validators.go (hunk 1 above)
Comment (kubawi) [reply aws/validators.go]: Good spot, thank you! Here is a relevant snippet from [Amazon MQ AWS API docs](https://docs.aws.amazon.com/amazon-mq/latest/api-reference/brokers.html#brokers-model-createbrokerinput) for reference:
> Required. The broker's name. This value must be unique in your AWS account, 1-50 characters long, must contain only letters, numbers, dashes, and underscores, and must not contain white spaces, brackets, wildcard characters, or special characters.

I've fixed the regex and added info to the docs in 228d529.

**Review by gdavison — APPROVED** [review]
Thanks for the PR, @kubawi, and apologies for the delay

Looks good! 🚀 

Acceptance test results

--- PASS: TestValidateBrokerName (0.00s)
--- PASS: TestBrokerPasswordValidation (0.00s)
--- PASS: TestDiffUsers (0.00s)
--- PASS: TestAccMQBroker_RabbitMQ_logs (712.66s)
--- PASS: TestAccMQBroker_RabbitMQ_validationAuditLog (776.08s)
--- PASS: TestAccMQBroker_throughputOptimized (868.59s)
--- PASS: TestAccMQBroker_RabbitMQ_basic (938.51s)
--- PASS: TestAccMQBroker_ldap (1049.03s)
--- PASS: TestAccMQBroker_RabbitMQ_cluster (1053.95s)
--- PASS: TestAccMQBroker_EncryptionOptions_awsOwnedKeyEnabled (1060.23s)
--- PASS: TestAccMQBrokerDataSource_basic (1095.74s)
--- PASS: TestAccMQBroker_basic (1100.66s)
--- PASS: TestAccMQBroker_Update_tags (1100.68s)
--- PASS: TestAccMQBroker_EncryptionOptions_awsOwnedKeyDisabled (1113.39s)
--- PASS: TestAccMQBroker_EncryptionOptions_kmsKeyID (1120.97s)
--- PASS: TestAccMQBroker_disappears (1139.20s)
--- PASS: TestAccMQBroker_Update_securityGroup (1331.22s)
--- PASS: TestAccMQBroker_Update_engineVersion (1348.04s)
--- PASS: TestAccMQBroker_Update_users (1483.20s)
--- PASS: TestAccMQBroker_Update_hostInstanceType (1695.80s)
--- PASS: TestAccMQBroker_AllFields_defaultVPC (1768.20s)
--- PASS: TestAccMQBroker_AllFields_customVPC (1814.28s)

**Pull request merged**